internal/trafficcontrol/tc.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trafficcontrol
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/vishvananda/netlink"
+)
+
+var (
+	sysClassNet = "/sys/class/net/"
+	tcpProtoStr = "tcp"
+	udpProtoStr = "udp"
+)
+
+func AllowTrafficFromMac(ifName string, macAddress string) error {
+	if _, err := exec.LookPath("tc"); err != nil {
+		return err
+	}
+	interfaces, err := GetAssociatedInterfaces(ifName)
+	if err != nil {
+		return err
+	}
+	for _, iface := range interfaces {
+		cmd := exec.Command("bash", "-c", fmt.Sprintf("tc filter replace dev %s ingress pref 9000 protocol all flower src_mac %s action ok", iface, macAddress))
+		err := cmd.Run()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func DenyTrafficFromMac(ifName string, macAddress string) error {
+	if _, err := exec.LookPath("tc"); err != nil {
+		return err
+	}
+	interfaces, err := GetAssociatedInterfaces(ifName)
+	if err != nil {
+		return err
+	}
+	for _, iface := range interfaces {
+		cmd := exec.Command("bash", "-c", fmt.Sprintf("tc filter replace dev %s ingress pref 9000 protocol all flower src_mac %s action drop", iface, macAddress))
+		err := cmd.Run()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func InitInterfaceForEAPTraffic(logger log.Logger, ifName string, unprotectTcpPorts, unprotectUdpPorts []int) error {
```

Comment:
Still not called anywhere, but I understand why.

I'd much rather have a second entrypoint or an `--init` flag in main.go to call this same code for the init container so all the 'tc' operations are in one file and not split across this file AND a shell script, but that can be in a separate PR.

Reply:
sure, will do this with another PR and that'll completely remove using scripts.